x/ccv/provider/types/genesis.go

```diff
@@ -33,28 +33,107 @@ func NewGenesisState(
 
 func DefaultGenesisState() *GenesisState {
 	return &GenesisState{
-		Params: DefaultParams(),
+		ValsetUpdateId: DefaultValsetUpdateID,
+		Params:         DefaultParams(),
 	}
 }
 
 func (gs GenesisState) Validate() error {
+	if gs.ValsetUpdateId == 0 {
+		return sdkerrors.Wrap(ccv.ErrInvalidGenesis, "valset update ID cannot be equal to zero")
+	}
+
+	for _, ubdOp := range gs.UnbondingOps {
+		if len(ubdOp.UnbondingConsumerChains) == 0 {
+			return sdkerrors.Wrap(ccv.ErrInvalidGenesis, "unbonding operations cannot have an empty consumer chain list")
+		}
+	}
+
+	for _, prop := range gs.ConsumerAdditionProposals {
+		if err := prop.ValidateBasic(); err != nil {
+			return sdkerrors.Wrap(ccv.ErrInvalidGenesis, err.Error())
+		}
+	}
+
+	for _, prop := range gs.ConsumerRemovalProposals {
+		if err := prop.ValidateBasic(); err != nil {
+			return sdkerrors.Wrap(ccv.ErrInvalidGenesis, err.Error())
+		}
+	}
+
+	if len(gs.ValsetUpdateIdToHeight) > 1 {
+		// check only the first tuple of the list since it is ordered by VSC ID
+		if gs.ValsetUpdateIdToHeight[0].ValsetUpdateId == 0 {
+			return sdkerrors.Wrap(ccv.ErrInvalidGenesis, "valset update ID cannot be equal to zero")
+		}
+	}
+
 	for _, cs := range gs.ConsumerStates {
 		if err := cs.Validate(); err != nil {
-			return err
+			return sdkerrors.Wrap(ccv.ErrInvalidGenesis, fmt.Sprintf("%s: for consumer chain id: %s", err, cs.ChainId))
 		}
 	}
+
 	if err := gs.Params.Validate(); err != nil {
 		return err
 	}
+
 	return nil
 }
 
+// Validate performs a consumer state validation returning an error upon any failure.
+// It ensures that the chain id, client id and consumer genesis states are valid and non-empty.
 func (cs ConsumerState) Validate() error {
-	if strings.TrimSpace(cs.ChainId) == "" {
-		return sdkerrors.Wrap(sdkerrors.ErrInvalidChainID, "consumer chain id cannot be empty string")
-	}
 	if err := host.ChannelIdentifierValidator(cs.ChannelId); err != nil {
-		return sdkerrors.Wrapf(err, "ccv channel id for chain %s is not valid", cs.ChainId)
+		return err
+	}
+	if err := host.ClientIdentifierValidator(cs.ClientId); err != nil {
+		return err
+	}
+	// consumer genesis should be for a new chain only
+	if !cs.ConsumerGenesis.NewChain {
+		return fmt.Errorf("consumer genesis must be for a new chain")
+	}
+	// validate a new chain genesis
+	if err := cs.ConsumerGenesis.Validate(); err != nil {
+		return err
+	}
+
+	// validate optional fields
+
+	if err := validateSlashAcksAddress(cs.SlashDowntimeAck); err != nil {
+		return err
+	}
+
+	for _, pVSC := range cs.PendingValsetChanges {
+		if pVSC.ValsetUpdateId == 0 {
+			return fmt.Errorf("valset update ID cannot be equal to zero")
+		}
+		if len(pVSC.ValidatorUpdates) == 0 {
```

Comment:
This is actually possible. See https://github.com/cosmos/interchain-security/blob/a8f854b777bc7c3bc0f1a2532dd084a8c10855fd/x/ccv/provider/keeper/relay.go#L196